test/test_images/logevents/main.go

```diff
@@ -20,13 +20,16 @@ import (
 	"time"
 
 	"github.com/knative/pkg/cloudevents"
-
-	corev1 "k8s.io/api/core/v1"
 )
 
-func handler(ctx context.Context, e *corev1.Event) {
+type Heartbeat struct {
+	Sequence int    `json:"id"`
+	Data     string `json:"data"`
+}
+
+func handler(ctx context.Context, hb *Heartbeat) {
```

Comment:
Not sure about this change - seems like it should be type-agnostic to accommodate future tests. Don't feel strongly though as long as it works (and it works, yay!)

I definitely like that this is its own thing so it can evolve as the e2e tests require without considering breakage (unlike the message-dumper tool).